dkist/io/asdf/tests/test_dataset.py

```diff
@@ -111,17 +111,17 @@ def test_save_dataset_with_file_schema(tagobj, tmpdir):
             afile.write_to(Path(tmpdir / "test.asdf"))
 
 
-@pytest.mark.parametrize("asdf_file",
-                         [rootdir / "eit_dataset_0.1.0.asdf",
-                          rootdir / "eit_dataset_0.2.0.asdf",
-                          rootdir / "eit_dataset_0.3.0.asdf"])
-def test_read_all_schema_versions(asdf_file):
+def test_read_all_schema_versions(eit_dataset_asdf_path):
     """
     This test validates that we can successfully read a full and valid Dataset
     object from files with all versions of the dataset schema.
     """
     with importlib_resources.as_file(importlib_resources.files("dkist.io") / "level_1_dataset_schema.yaml") as schema_path:
-        with asdf.open(asdf_file) as afile:
+        # Firstly verify that the tag versions in the test filename are the ones used in the file
+        with asdf.open(eit_dataset_asdf_path, _force_raw_types=True) as afile:
+            assert afile["dataset"]._tag.rsplit("/")[-1] in str(eit_dataset_asdf_path)
```

Comment:
Relying on the path to a test file seems a bit fragile, but w/e.

Reply:
Yeah, the alternative was a lot more work!